Introduce isPast(), isFuture() and isInfinite() on DateRanges and DateTimeRanges

rquadling opened the pull request:
Allow users to determine if a date/datetime range is anchored in the past, the future or both.

gws commented on library/Vo/DateRange.php:
```diff
@@ -476,4 +476,34 @@ public function __toString()
+     */
+    public function isInfinite()
+    {
+        return self::infinite() == $this;
```
I think this should be `static::` instead of `self::`, in case `infinite()` is overridden by a subclass. It's also consistent with the `isFuture()` and `isPast()`.

gws commented:
Thank you! If you want to resolve that small change I suggested and rebase I can merge this.

rquadling commented:
How do you rebase in github?

gws commented:
@rquadling Hmm, I just meant to squash the commit. You can squash commits interactively using `git rebase -i` on the branch you're requesting that I merge, and then force-push. GitHub will see that you've done this and this PR will be updated automatically.

`git rebase -i HEAD~1` in your case.

https://ariejan.net/2011/07/05/git-squash-your-latests-commits-into-one/

Commit:
Introduce isPast(), isFuture() and isInfinite() on DateRanges and DateTimeRanges

Allow users to determine if a date/datetime range is anchored in the past, the future or both.

rquadling commented:
Tada!

gws commented:
Thank you, and I've released v0.4.0 with these changes.
